ocs_ci/ocs/ocs_upgrade.py

```diff
@@ -385,3 +391,124 @@ def set_upgrade_images(self):
             with NamedTemporaryFile() as cs_yaml:
                 dump_data_to_temp_yaml(cs_data, cs_yaml.name)
                 ocs_catalog.apply(cs_yaml.name)
+
+
+class UpgradeDisruptiveOps(EcosystemTest):
+    """
+    Contains disruptive upgrade tests functions
+
+    """
+
+    def get_ocs_operator_node_name(self):
+        """
+        Getting node's name that running ocs-operator pod
+
+        Returns:
+            str: node's name that running ocs-operator pod
+
+        """
+        ocs_operator_pod = get_ocs_operator_pod()
+        log.debug(f"ocs operator pod info: {ocs_operator_pod}")
+        ocs_operator_node = get_pod_node(ocs_operator_pod)
+
+        return get_node_name(ocs_operator_node)
+
+    def worker_node_shutdown(self, nodes, abrupt):
+        """
+        Shutdown worker node that running ocs-operator pod
+
+        Args:
+            abrupt: bool: True if abrupt shutdown, False for permanent shutdown
+
+        """
+        log.info(f"Abrupt {abrupt}")
+        # get ocs-operator node:
+        ocs_operator_node_name = self.get_ocs_operator_node_name()
+        # get rook-ceph-tools node:
+        rook_ceph_tools_pod = get_ceph_tools_pod()
+        rook_ct_node = get_pod_node(rook_ceph_tools_pod)
+        rook_ct_node_name = get_node_name(rook_ct_node)
+        if rook_ct_node_name == ocs_operator_node_name:
+            log.info("rook ceph tools pod and ocs-operator pod running on same node."
+                     "deleting ocs-operator pod")
+            delete_pods([get_ocs_operator_pod()])
+            time.sleep(120)
+            ocs_operator_node_name = self.get_ocs_operator_node_name()
+            assert not (ocs_operator_node_name == rook_ct_node_name), (
+                "rook-ceph-tools and ocs-operator pods running on same node"
+            )
+
+        # get workers node objects:
+        node_to_shutdown = list()
+        for node in get_typed_nodes():
+            node_name = get_node_name(node)
+            log.info(f"node: {node_name}, ocs operator node: {ocs_operator_node_name}")
+            if node_name == ocs_operator_node_name:
+                if node_name != rook_ct_node_name:
```

Comment:
consider using tests.helpers.wait_for_ct_pod_recovery so you will not need to exclude the node running the CT pod.
See reference in tests/manage/z_cluster/nodes/test_disk_failures.py:138